libs/langchain/langchain/schema/runnable.py

```diff
@@ -186,6 +188,110 @@ def _call_with_config(
             )
             return output
 
+    def with_fallbacks(
+        self,
+        fallbacks: Sequence[Runnable],
+        exceptions_to_handle: Sequence[Type[BaseException]] = (
+            KeyboardInterrupt,
+            Exception,
+        ),
+    ) -> RunnableWithFallbacks:
+        return RunnableWithFallbacks(
+            runnable=self,
+            fallbacks=fallbacks,
+            exceptions_to_handle=exceptions_to_handle,
+        )
+
+
+class RunnableWithFallbacks(Serializable, Runnable[Input, Output]):
+    runnable: Runnable[Input, Output]
+    fallbacks: Sequence[Runnable[Input, Output]]
+    exceptions_to_handle: Sequence[Type[BaseException]] = (KeyboardInterrupt, Exception)
```

Comment:
i feel like it should NOT handle Keyboard Interrupt by default? I feel like that means you want to cancel the whole thing

Reply:
Yea I'd tend to agree

Reply:
👍